commands/tocfl/chewing.py

```diff
@@ -0,0 +1,262 @@
+}
+
+# In Hanyu Pinyin, 「ㄜ」 and 「ㄝ」 share the same character 〔e〕
+# because the possible initials are almost different.
+# Only 「誒」 (「ㄝˋ」) interferes with 「惡」(「ㄜˋ」),
+# in this case 「ㄝ」 is spelt as 〔ê〕
+PINYIN_ALONE = {
+    "zhi": "ㄓ",
+    "chi": "ㄔ",
+    "shi": "ㄕ",
+    "ri": "ㄖ",
+    "zi": "ㄗ",
+    "ci": "ㄘ",
+    "si": "ㄙ",
+    "a": "ㄚ",
+    "o": "ㄛ",
+    "e": "ㄜ",
+    "ê": "ㄝ",
+    "ai": "ㄞ",
+    "ei": "ㄟ",
+    "ao": "ㄠ",
+    "ou": "ㄡ",
+    "an": "ㄢ",
+    "en": "ㄣ",
+    "ang": "ㄤ",
+    "er": "ㄦ",
+    # In case of Erhua, 〔r〕 is adeed to the last character instead of 〔er〕
+    "r": "ㄦ",
+    "yi": "ㄧ",
+    "wu": "ㄨ",
+    "yu": "ㄩ",
+    # Combined
+    "ya": "ㄧㄚ",
+    "yo": "ㄧㄛ",
+    "ye": "ㄧㄝ",
+    "yai": "ㄧㄞ",
+    "yao": "一ㄠ",
+    "you": "ㄧㄡ",
+    "yan": "ㄧㄢ",
+    "yin": "ㄧㄣ",
+    "yang": "ㄧㄤ",
+    "ying": "ㄧㄥ",
+    "wa": "ㄨㄚ",
+    "wo": "ㄨㄛ",
+    "wai": "ㄨㄞ",
+    "wei": "ㄨㄟ",
+    "wan": "ㄨㄢ",
+    "wen": "ㄨㄣ",
+    "wang": "ㄨㄤ",
+    "weng": "ㄨㄥ",
+    "yue": "ㄩㄝ",
+    "yuan": "ㄩㄢ",
+    "yun": "ㄩㄣ",
+    "yong": "ㄩㄥ",
+    # v is used to replace ü in typing
+    "nü": "ㄋㄩ",
+    "lü": "ㄌㄩ",
+    "nv": "ㄋㄩ",
+    "lv": "ㄌㄩ",
+}
+
+PINYIN_CENTER = {
+    "i": "ㄧ",
+    "u": "ㄨ",
+    "ü": "ㄩ",
+    "v": "ㄩ",
+}
+
+# The designer of Hanyu Pinyin used e to represent both 「ㄜ」 and 「ㄝ」.
+# This is because 「ㄝ」 could only be used in  「ㄩㄝ」 and 「ㄧㄝ」
+PINYIN_FINALS = {
+    "a": "ㄚ",
+    "o": "ㄛ",
+    "e": "ㄜ",  # also ㄝ
+    "ai": "ㄞ",
+    "ei": "ㄟ",
+    "ao": "ㄠ",
+    "ou": "ㄡ",
+    "an": "ㄢ",
+    "en": "ㄣ",
+    "ang": "ㄤ",
+    "eng": "ㄥ",
+    "er": "ㄦ",
+}
+
+PINYIN_COMBINED = {
+    "iu": "ㄧㄡ",
+    "ian": "ㄧㄢ",
+    "in": "ㄧㄣ",
+    "iang": "ㄧㄤ",
+    "ing": "ㄧㄥ",
+    "ui": "ㄨㄟ",
+    "uan": "ㄨㄢ",  # also ㄩㄢ
+    "un": "ㄨㄣ",  # also ㄨㄣ
+    "uang": "ㄨㄤ",
+    "ong": "ㄨㄥ",
+    "ue": "ㄩㄝ",
+    "iong": "ㄩㄥ",
+}
+
+# Only 「a﹑o﹑e﹑i﹑u﹑ü」 are added diacritics in Hanyu Pinyin
+# Neutral tones are not labeled
+DIACRITIC_TO_BASE_AND_TONE = {
+    # ā (ɑ̄) ē ī ō ū ǖ
+    # á (ɑ́) é í ó ú ǘ
+    # ǎ (ɑ̌) ě ǐ ǒ ǔ ǚ
+    # à (ɑ̀) è ì ò ù ǜ
+    "ā": ("a", "¯"),
+    "á": ("a", "ˊ"),
+    "ǎ": ("a", "ˇ"),
+    "ă": ("a", "ˇ"),
+    "à": ("a", "ˋ"),
+    "ē": ("e", "¯"),
+    "é": ("e", "ˊ"),
+    "ě": ("e", "ˇ"),
+    "ĕ": ("e", "ˇ"),
+    "è": ("e", "ˋ"),
+    "ī": ("i", "¯"),
+    "í": ("i", "ˊ"),
+    "ǐ": ("i", "ˇ"),
+    "ĭ": ("i", "ˇ"),
+    "ì": ("i", "ˋ"),
+    "ō": ("o", "¯"),
+    "ó": ("o", "ˊ"),
+    "ǒ": ("o", "ˇ"),
+    "ŏ": ("o", "ˇ"),
+    "ò": ("o", "ˋ"),
+    "ū": ("u", "¯"),
+    "ú": ("u", "ˊ"),
+    "ǔ": ("u", "ˇ"),
+    "ŭ": ("u", "ˇ"),
+    "ù": ("u", "ˋ"),
+    "ǖ": ("ü", "¯"),
+    "ǘ": ("ü", "ˊ"),
+    "ǚ": ("ü", "ˇ"),
+    "ü̆": ("ü", "ˇ"),
+    "ǜ": ("ü", "ˋ"),
+    "ề": ("ê", "ˋ"),  # 「ㄝ」 could only possibly be the fourth tone
+}
+
+
+def to_chewing(pinyin: str) -> str:
+
+    # Matches chewing from substrings
+    def match_chewing(string: str, index: int, target: dict[str, str]):
```

Comment:
`match_chewing` could be moved outside of `to_chewing` to reduce clutter, since defining functions inside functions usually means you want to make decorators (which is not the case here)